chapters/dump-reload.rst

```diff
@@ -0,0 +1,127 @@
+for *querying* data. **Write-once** means that the data is only written
+to the indexer once, at initialization time. After that, you can only
+issue ``search`` requests to it. In order to re-build it with new data,
+you need to use the ``.rolling_update`` method.
+
+Dump-Reload
+^^^^^^^^^^^
+
+The Flow object provides two methods, ``.dump`` and ``.rolling_update``,
+for configuring the rolling updates.
+
+``.rolling_update`` should be called on the Query Flow, with the name of
+the Pod where the Indexer resides, and the path from which to read. The
+method then shuts down the indexers in the Flow sequentially, and
+instructs them to start again, but with the new data from the location
+provided.
+
+.. code:: python
+
+   with flow_query:
+       flow_query.rolling_update('indexer_query', dump_path)
+
+``.dump`` should be called on the DBMS (Indexing) Flow, with the name of
+the Pod where the Indexer resides, which path to dump to, and how many
+`shards <https://docs.jina.ai/chapters/parallel/#intra-parallelism>`__
+you have in your Query Flow.
+
+.. code:: python
+
+   with flow_dbms:
+       flow_dbms.dump('indexer_dbms', dump_path=dump_path, shards=shards)
+
+The method then extracts the data from the DBMS Indexer and stores it in
+the specified location, prepared to be read by Query Indexers.
+
+You can see a graphical representation of the process:
+
+.. image:: images/dump-reload/dump-reload.jpg
+
+
+Replicas and rolling updates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Replicas are, like the name suggests, duplicates of a specific Pod. They
+are parallel, so a query request will have ``polling: any`` between the
+two of them. These are required on the Query Flow side in order to
+maintain availability during the ``.rolling_update`` call. You need to
+make sure that you have at least 2 replicas configured for your Query
+Indexer. This way, when one is taken offline for reloading, the other
+one is still alive and serving results.
+
+This can be configured via the Python API, or in the YAML of the Pod:
+
+.. code:: yml
+
+   ...
+   pods:
+     - name: indexer_query
+       uses: indexer_query.yml
+       replicas: 2
+   ...
+
+You can see a graphical representation of the rolling update process below:
+
+.. image:: images/dump-reload/replicas.jpg
+
+In this graphic ``Dump Time 1`` and ``Dump Time 2`` represent the inner state of the two replicas. One is serving data from ``Time 1`` (with continuous line), while the other is being reloaded with new data (from ``Time 2``, dotted line).
+
+JinaD
+'''''
+
+When running remote Flows, you can access these methods via
+``PUT /flows/{flow_id}``. You can check the endpoint docs
+`here <https://api.jina.ai/daemon/#operation/_update_flows__id__put>`__
+
+Example
+^^^^^^^
+
+To see how this would be used in an actual application, check out our example `here <https://github.com/jina-ai/examples/tree/master/wikipedia-sentences-query-while-indexing>`_
```

Comment:
@FionnD is this a separate example or will we merge it into the wiki example like we did incremental indexing?

Reply:
Separate as per discussion here https://jinaai.slack.com/archives/C018NC4SYAZ/p1620801607122900

Reply:
We won't, since it's radically different. The incr indexing is v similar.